src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/MigrateVisitService.kt

```diff
@@ -57,56 +61,149 @@ class MigrateVisitService(
   @Autowired
   private lateinit var visitDtoBuilder: VisitDtoBuilder
 
+  @Autowired
+  private lateinit var sessionSlotService: SessionSlotService
+
+  @Autowired
+  private lateinit var applicationRepository: ApplicationRepository
+
   fun migrateVisit(migrateVisitRequest: MigrateVisitRequestDto): String {
     if (isVisitTooFarInTheFuture(migrateVisitRequest.startTimestamp)) {
-      throw VisitToMigrateException("OldVisit more than $migrateMaxMonthsInFuture months in future, will not be migrated!")
+      throw VisitToMigrateException("Visit more than $migrateMaxMonthsInFuture months in future, will not be migrated!")
     }
 
     val actionedBy = migrateVisitRequest.actionedBy ?: NOT_KNOWN_NOMIS
     // Deserialization kotlin data class issue when OutcomeStatus = json type of null defaults do not get set hence below code
     val outcomeStatus = migrateVisitRequest.outcomeStatus ?: OutcomeStatus.NOT_RECORDED
 
+    val shouldMigrateWithSessionTemplate = shouldMigrateWithSessionMapping(migrateVisitRequest)
     val prison: Prison
-    var sessionTemplateReference: String ? = null
     val visitRoom: String
+    var sessionSlot: SessionSlot ? = null
 
-    if (shouldMigrateWithSessionMapping(migrateVisitRequest)) {
+    if (shouldMigrateWithSessionTemplate) {
       val sessionTemplate = migrationSessionTemplateMatcher.getMatchingSessionTemplate(migrateVisitRequest)
 
       prison = sessionTemplate.prison
-      sessionTemplateReference = sessionTemplate.reference
+      val sessionTemplateReference = sessionTemplate.reference
       visitRoom = sessionTemplate.visitRoom
+
+      sessionSlot = sessionSlotService.getSessionSlot(
+        startTimeDate = migrateVisitRequest.startTimestamp,
+        endTimeAndDate = migrateVisitRequest.endTimestamp,
+        sessionTemplateReference,
+        prison,
+      )
     } else {
       prison = prisonsService.findPrisonByCode(migrateVisitRequest.prisonCode)
       visitRoom = migrateVisitRequest.visitRoom
+
+      sessionSlot = sessionSlotService.getSessionSlot(
+        startTimeDate = migrateVisitRequest.startTimestamp,
+        endTimeAndDate = migrateVisitRequest.endTimestamp,
+        prison,
+      )
+    }
+
+    val applicationEntity: Application = createApplication(migrateVisitRequest, prison, sessionSlot, actionedBy)
```

Comment:
do we need to create an application for migrated visits?

Reply:
Yet's keep changes to minimal...